lang/funcs/simple/env_func.go

```diff
@@ -0,0 +1,76 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// functions to look up variables in the mgmt server instance environment
+
+package simple // TODO: should this be in its own individual package?
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+// getenv(str) str # get env or empty string if absent
+func getenv(input []types.Value) (types.Value, error) {
+	value := os.Getenv(input[0].Str())
+	return &types.StrValue{
+		V: fmt.Sprintf("%s", value),
+	}, nil
+}
+
+// defaultenv(str, str) str # get env var if exists or default value
+func defaultenv(input []types.Value) (types.Value, error) {
+	value, exists := os.LookupEnv(input[0].Str())
+	if !exists {
+		value = input[1].Str()
+	}
+	return &types.StrValue{
+		V: fmt.Sprintf("%s", value),
+	}, nil
+}
+
+// hasenv(str) bool # does env var exist?
+func hasenv(input []types.Value) (types.Value, error) {
+	_, exists := os.LookupEnv(input[0].Str())
+	return &types.BoolValue{
+		V: exists,
+	}, nil
+}
+
+// env() {str: str} # return a map of all keys and their values
+func env(input []types.Value) (types.Value, error) {
+	environ := make(map[types.Value]types.Value)
+	for _, keyval := range os.Environ() {
+		if i := strings.IndexRune(keyval, '='); i != -1 {
+			environ[&types.StrValue{V: keyval[:i]}] = &types.StrValue{V: keyval[i+1:]}
+		}
+	}
+	return &types.MapValue{
+		T: types.NewType("{str: str}"),
+		V: environ,
+	}, nil
+}
+
+func init() {
```

Comment:
even though it's not intuitive, init should be at the top where it was before. This is so you don't accidentally forget it's there and running code when you didn't expect it to :)